clang/test/CodeGenCXX/default-arguments.cpp

```diff
@@ -12,6 +12,31 @@ void g() {
 }
 }
 
+namespace GH113324 {
+struct S1 {
+  friend void f1(S1, int = 42) {}
+};
+
+template <bool, class> using __enable_if_t = int;
```

Comment:
I think you should at least remove these unrelated things in the test: they contributed nothing to the previous bug.

The reduced case is enough to demonstrate where we messed things up

```cpp
struct CacheRequestImpl {
  friend void LoadOrRun(CacheRequestImpl, int = 42) {}
};
void CacheRequestTests_MakesCacheKey_TestTestBody() {
  CacheRequestImpl req;
  LoadOrRun(req);
}
```

Reply:
@zyn0217 I've removed unnecessary tests. thanks